pkg/listeners/listeners_solaris.go

```diff
@@ -0,0 +1,62 @@
+package listeners
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/docker/go-connections/sockets"
+	"github.com/docker/libnetwork/portallocator"
+)
+
+// Init creates new listeners for the server.
+func Init(proto, addr, socketGroup string, tlsConfig *tls.Config) (ls []net.Listener, err error) {
+	switch proto {
+	case "tcp":
+		l, err := sockets.NewTCPSocket(addr, tlsConfig)
+		if err != nil {
+			return nil, err
+		}
+		ls = append(ls, l)
+	case "unix":
+		l, err := sockets.NewUnixSocket(addr, socketGroup)
+		if err != nil {
+			return nil, fmt.Errorf("can't create unix socket %s: %v", addr, err)
+		}
+		ls = append(ls, l)
+	default:
+		return nil, fmt.Errorf("Invalid protocol format: %q", proto)
+	}
+
+	return
+}
+
+// allocateDaemonPort ensures that there are no containers
```

Comment:
I'm not sure this one is on the right place ? It's unexported and doesn't seems to be used in the package (which is composed only by `listeners` files for `unix`, `windows` and this one `solaris`). So it feels like dead code to me..